### PR TITLE
refactor(core): Move tag collection into repository (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**

### CODE REVIEW

File: packages/cli/src/Server.ts [hunk 1]
```diff
@@ -481,7 +481,7 @@ export class Server extends AbstractServer {
 				logger,
 				jwtService,
 			}),
-			new TagsController({ config, repositories, externalHooks }),
+			new TagsController({ config, tagRepository: Container.get(TagRepository), externalHooks }),
```
Comment (netroy) [inline packages/cli/src/Server.ts:484]: maybe we can import `config` directly in `TagsController`, add `@Service()` to that class, and replace this line with `Container.get(TagsController)`. 
That's the eventual goal: to stop explicitly passing in dependencies everywhere.

File: packages/cli/test/integration/shared/testDb.ts [hunk 2]
```diff
@@ -113,7 +115,13 @@ export async function terminate() {
  * Truncate specific DB tables in a test DB.
  */
 export async function truncate(collections: CollectionName[]) {
-	for (const collection of collections) {
+	const [tag, rest] = separate(collections, (c) => c === 'Tag');
```
Comment (netroy) [inline packages/cli/test/integration/shared/testDb.ts:118]: as we remove repositories from `Db.collections`, maybe we should switch to each test file defining what tables **not to truncate**, and then we can truncate all other tables.

File: packages/cli/test/integration/shared/utils/testServer.ts [hunk 3]
```diff
@@ -275,7 +276,11 @@ export const setupTestServer = ({
 						registerController(
 							app,
 							config,
-							new TagsController({ config, externalHooks, repositories }),
+							new TagsController({
```
Comment (netroy) [inline packages/cli/test/integration/shared/utils/testServer.ts:279]: same as mentioned above: if we make `TagsController` a `@Service`, can replace this block with `Container.get(TagsController)`

File: packages/cli/src/TagHelpers.ts [hunk 4]
```diff
@@ -1,6 +1,8 @@
 import type { EntityManager } from 'typeorm';
```
Comment (netroy) [inline packages/cli/src/TagHelpers.ts:1]: almost all of the code in this file belongs to the `TagRepository` file.

File: packages/cli/src/environments/sourceControl/sourceControl.service.ee.ts [hunk 5]
```diff
@@ -682,7 +682,7 @@ export class SourceControlService {
 		options: SourceControlGetStatus,
 		sourceControlledFiles: SourceControlledFile[],
 	) {
-		const lastUpdatedTag = await Db.collections.Tag.find({
+		const lastUpdatedTag = await Container.get(TagRepository).find({
```
Comment (netroy) [inline packages/cli/src/environments/sourceControl/sourceControl.service.ee.ts:685]: you can inject this in the constructor instead.

File: packages/cli/src/environments/sourceControl/sourceControlExport.service.ee.ts [hunk 6]
```diff
@@ -167,7 +168,7 @@ export class SourceControlExportService {
 	async exportTagsToWorkFolder(): Promise<ExportResult> {
 		try {
 			sourceControlFoldersExistCheck([this.gitFolder]);
-			const tags = await Db.collections.Tag.find();
+			const tags = await Container.get(TagRepository).find();
```
Comment (netroy) [inline packages/cli/src/environments/sourceControl/sourceControlExport.service.ee.ts:171]: can be injected in the constructor

File: packages/cli/src/environments/sourceControl/sourceControlImport.service.ee.ts [hunk 7]
```diff
@@ -265,7 +266,7 @@ export class SourceControlImportService {
 		tags: TagEntity[];
 		mappings: WorkflowTagMapping[];
 	}> {
-		const localTags = await Db.collections.Tag.find({
+		const localTags = await Container.get(TagRepository).find({
```
Comment (netroy) [inline packages/cli/src/environments/sourceControl/sourceControlImport.service.ee.ts:269]: constructor injection here as well

**Review by netroy — CHANGES_REQUESTED** [review]
besides minor comments, LGTM.

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.2.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.2.1)

**Pull request merged**